datafusion/sql/src/unparser/expr.rs

```diff
@@ -221,9 +216,10 @@ impl Unparser<'_> {
                         ast::WindowFrameUnits::Groups
                     }
                 };
-                let order_by: Vec<ast::OrderByExpr> = order_by
+
+                let order_by =order_by
                     .iter()
-                    .map(sort_to_sql)
+                    .map(|sort_expr| self.sort_to_sql(sort_expr))
```

Comment:
Use the right self.sort_to_sql function that respects `supports_nulls_first_in_sort`